datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -168,6 +164,22 @@ protected AbstractKafkaBasedConnectorTask(KafkaBasedConnectorConfig config, Data
     _startOffsets = Optional.ofNullable(_datastream.getMetadata().get(DatastreamMetadataConstants.START_POSITION))
         .map(json -> JsonUtils.fromJson(json, new TypeReference<Map<Integer, Long>>() {
         }));
+
+    // if this datastream want to use specific start offsets, we need to have the auto.offset.reset config
+    // set to "none" in the kafka consumer configs, so that the Kafka consumer throws NoOffsetForPartiionException
+    // upon first poll, to be handled by seeking to teh startOffsets
+    if (_startOffsets.isPresent()) {
```

Comment:
unrelated to your change, but should we make _startOffsets not Optional? It is a map and we are wrapping it in a Nullable only to be using isPresent everywhere if present needing to access it with the get call which is not serving the purpose of Optional at all.

Reply:
makes sense. I made _startOffsets not Optional